azurerm/internal/services/network/public_ip_resource.go

```diff
@@ -209,6 +218,25 @@ func resourcePublicIpCreateUpdate(d *schema.ResourceData, meta interface{}) erro
 		publicIp.PublicIPAddressPropertiesFormat.DNSSettings = &dnsSettings
 	}
 
+	if v, ok := d.GetOk("ip_tags"); ok {
+		ipTags := v.(map[string]interface{})
+		newIpTags := []network.IPTag{}
+
+		for key, val := range ipTags {
+			if strings.EqualFold(key, "routingpreference") && !strings.EqualFold(sku, "standard") {
```

Comment:
I'm hesitant to put this style of error here as it can cause issues with maintainability if we have to keep coming back and adding to it. We might as well let the api handle it. Do you know if it is just specific types of tags that are forced to be on the standard sku or is it all tags?

Reply:
Do you also know if the error message from the api is clear that `RoutingPreference` required the Standard SKU?

Reply:
Agreed. The error message is clear so I have removed this.